### DISCUSSION TITLE
workflowsdkstepenabled is false in target environment

### DISCUSSION THREAD

**Discussion by MudlThru** [drivardxrm/Dataverse-CustomApis#6] [General]
You mention in your [article](https://itmustbecode.com/dataverse-custom-api-enabled-for-workflow/) that the customapis can be called from a classic workflow.

Did you encounter the issue when deploying your apis to a new environment where the sdkmessage workflowsdkstepenabled field is defaulted to false in the target environment? If so, how did you get around it?

**Comment by drivardxrm** [comment]
Hi @MudlThru ,
Unfortunately, I only had problems working with the workflowsdkstepenabled feature. It works well at design time, but the experience is a complete mess when it comes to transporting the Apis to other environments.

I think there was a blog post on the topic, but I can't find it.

Personally,  when need to use custom api logic in a workflow, I create a custom action with the same inputs and outputs and I add a plugin step on the custom api plugin that triggers on the custom action.

It sucks, but it's the best solution I found so far.

Good luck!

**Comment by peta** [comment]
@MudlThru @drivardxrm Same here - at least I am glad to have found this github issue here where others confirm my observations. While google yields loads of happy path articles how things *should* work, no reports about issues with it. I have instances where it doesn't even work at design-time. It seems that the Custom API (to be more precise the `sdkmessage` record being auto-magically generated in the background by Dv) sometimes als "breaks" when updating it, like adding/removing a request or response parameter. 

Did you already create a MS ticket for it?

It's a pity, that even as of today the shiny new Dataverse feature "Custom APIs" is rather a shaky preview-feature than a solid alternative to legacy Actions (which also had their flaws for many years now).
